theopensystemslab/planx-new
# feat: generate the BOPS payload on the backend from a sessionId

Depends on https://github.com/theopensystemslab/planx-core/pull/21

Changes:
- [x] adds new endpoint: GET `/admin/session/:sessionId/bops`
- [x] updates POST `/bops/:localAuthority` to only require a `sessionId` to be passed through from `createSendEvents`, no longer the full BOPS payload

Next steps (followup PR I think to keep changes more focused here & because CSV touches Uniform which has longer testing turn-around than BOPS?):
- refactor final reference to `getBOPSParams` in `makeCsvData` method in the editor (see `.../Send/uniform/index`) to use planx-core
  - I was having a hard time calling `_public.generateBOPSPayload` in the editor; what are the existing examples of using `CoreDomainClient` methods in editor?? this should ultimately move to API anyways, so I can just do that, but am curious
- remove `src/@planx/components/Send/bops` subdirectly & `../Send/model` BOPS-related types in editor, rely entirely on imports from planx-core

To test: 
- confirm you can submit an LDC app to BOPS staging successfully! 
  - flow: https://1671.planx.pizza/buckinghamshire/apply-for-a-lawful-development-certificate/preview
  - event should only have `sessionId` in payload and succeed: https://hasura.1671.planx.pizza/console/events/one-off-scheduled-events/processed
  - audit table should have a success record: https://hasura.1671.planx.pizza/console/data/default/schema/public/tables/bops_applications/browse
